src/App.js

```diff
@@ -427,6 +426,10 @@ export default class App extends Base {
           if (query.params.pubSubUrl) {
             process.env.PUB_SUB_URL = query.params.pubSubUrl;
           }
+          // Set the pubSub token if present
+          if (query.params.pubSubToken) {
+            process.env.PUB_SUB_TOKEN = query.params.pubSubToken;
+          }
 
           if (query.task) {
```

Comment:
Ability to accept SAT (as FCA is open source use pubSubToken) in FCA

In intent via Discovery.navigateTo or Parameters.initialization
in urlParams


all 3 points are not addressed

Reply:
in URL https://github.com/comcast-firebolt/fca-parent/pull/414/files#diff-e41fe23dd609642313d2b1baec982c44f3e2c3cbfe5d42f4d23f7813c65118b1R152 , is this the value assigned ?

Reply:
Regarding 1st point, 
both parameters.initialisation and discovery.navigateto is handled inside this same function if scrolled up
```
 async getParameterInitializationValues() {
    await handleAsyncFunction(FireboltExampleInvoker.get().invoke(CONSTANTS.CORE.toLowerCase(), 'Parameters.initialization', [], [])).then((res) => {
      console.log('Response of Initialization :: ', res);
      if (res != undefined) {
        const action = res[0].discovery.navigateTo.action;
        if (action == 'search') {
          let query = res[0].discovery.navigateTo.data.query;
```
There is no other place in fca where discovery.navigateto is called to extract intent params. I cross-checked with Preethi, same is the way support to get pubSubUrl in intent was added
As part of this ticket, support was added to get pubSubToken via intent as support to get pubSubUrl is already implemented

Regarding point 2, will confirm after testing

Reply:
discovery.onNavigateTo is not called in the above function. Only parameters.initialization


parameters.initialization is for cold launch
discovery.onNavigateTo is for hot launch 
so if you see here, https://github.com/rdkcentral/firebolt-certification-app/blob/f8e67883d38b4d9adbde1d8fe6ce99332af9e8f6/src/LifeCycleHistory.js#L102 , we are listening to this event and extracting values from message 
https://github.com/rdkcentral/firebolt-certification-app/blob/f8e67883d38b4d9adbde1d8fe6ce99332af9e8f6/src/LifeCycleHistory.js#L113

Reply:
Have added the logic to extract and store sat here as well..but i do see this logic is executed if testoken present
 ```
if (query.params && query.params.appId && query.params.testtoken && query.params.macaddress) {
            // PUBSUB_CONNECTION environment variable has a pubsub client instance and calls the isConnected function to check the Websocket status.
            if (!process.env.PUBSUB_CONNECTION || (process.env.PUBSUB_CONNECTION && !process.env.PUBSUB_CONNECTION.isConnected())) {
              process.env.APP_TYPE = query.params.appType ? query.params.appType.toLowerCase() : CONSTANTS.FIREBOLT_CONST;
              process.env.CURRENT_APPID = query.params.appId;
              process.env.MACADDRESS = query.params.macaddress;
              process.env.TEST_TOKEN = query.params.testtoken;
              process.env.PUB_SUB_TOKEN = query.params.pubSubToken;
              const pubSubListenerCreation = new PubSubCommunication();
              const webSocketConnection = await pubSubListenerCreation.startWebSocket();
            }
          }
```
Not exactly sure if testoken is passed everytime in intent or is a required param here..so what happens if testtoken is not passed in intent..as in do we need to move the logic to get sat outside this if and place it seperately below this logic?

Reply:
Update - Removed testtoken logic as discussed